src/Fizzics/level2.js

```diff
@@ -19,16 +21,90 @@ var FizzicsLevel2 = GObject.registerClass({
         this._content.add(this._codeview);
     }
 
-    static _toScopeName(name) {
-        return name.replace(/-/g, '_');
+    _getPropsForGlobals() {
+        void this;
+        return {
+            background: 'backgroundImageIndex',
+            showMoveTool: 'moveToolActive',
+            showFlingTool: 'flingToolActive',
+            showCreateTool: 'createToolActive',
+            showDeleteTool: 'deleteToolActive',
+        };
+    }
+
+    _getPropsForIndex(index) {
+        void this;
+        return {
+            radius: `radius_${index}`,
+            gravity: `gravity_${index}`,
+            collision: `collision_${index}`,
+            friction: `friction_${index}`,
+            lock: `usePhysics_${index}`,
+            attraction0: `socialForce_${index}_0`,
+            attraction1: `socialForce_${index}_1`,
+            attraction2: `socialForce_${index}_2`,
+            attraction3: `socialForce_${index}_3`,
+            attraction4: `socialForce_${index}_4`,
+            skin: `imageIndex_${index}`,
+            visualBad: `deathVisualBad_${index}`,
+            soundBad: `deathSoundBad_${index}`,
+            visualGood: `deathVisualGood_${index}`,
+            soundGood: `deathSoundGood_${index}`,
+        };
+    }
+
+    _createScopeWithProps(props) {
+        void this;
+        const scope = {};
+        Object.keys(props).forEach(key => {
+            scope[key] = null;
+        });
+        return scope;
     }
 
-    static _toModelName(name) {
-        return name.replace(/_/g, '-');
+    _createScopeForObject(index) {
+        const props = this._getPropsForIndex(index);
+        return this._createScopeWithProps(props);
     }
 
-    _getModelProps() {
-        return GObject.Object.list_properties.call(this._model.constructor.$gtype);
+    _createScope() {
+        const props = this._getPropsForGlobals();
+        const scope = this._createScopeWithProps(props);
+        scope.objects = Array.from({length: OBJECTS_NUM}).map((value, index) => {
+            return this._createScopeForObject(index);
+        });
+        return scope;
+    }
+
+    _updateModel(scope, props) {
+        Object.keys(props).forEach(key => {
+            const prop = props[key];
+            if (scope[key] === null)
+                return;
+
+            /* handle UI customizations here */
+            var scope_value = scope[key];
+            if (key === 'lock')
+                scope_value = !scope_value;
```

Comment:
I'm not really sure what this block achieves, despite the comment.

Reply:
Like you mention in another comment, I need to move these transformations into a common place if possible. These exists to adjust what the UI design is VS what the actual game-code (and thus, our proxy model) does.